systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java

```diff
@@ -236,15 +244,59 @@ public static String generateRandomConsumerGroup() {
         return CONSUMER_GROUP_NAME + salt;
     }
 
-    public static KafkaClientsBuilder getDefaultClientBuilder(TestStorage testStorage) {
+    public static KafkaClients generateNewConsumerGroup(KafkaClients clients) {
+        final String newConsumerGroup = ClientUtils.generateRandomConsumerGroup();
+        LOGGER.info("Regenerating new consumer group {} for clients {} {}", newConsumerGroup, clients.getProducerName(), clients.getConsumerGroup());
+
+        return new KafkaClientsBuilder(clients)
+            .withConsumerGroup(newConsumerGroup)
+            .build();
+    }
+
+    // instant client builders
+
+    private static KafkaClientsBuilder instantClientBuilderBase(TestStorage testStorage) {
         return new KafkaClientsBuilder()
-            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()))
             .withNamespaceName(testStorage.getNamespaceName())
-            .withMessageCount(testStorage.getMessageCount())
-            .withUsername(testStorage.getUsername())
+            .withMessageCount(testStorage.getMessageCount()) // default 100
+            .withDelayMs(0)
             .withTopicName(testStorage.getTopicName())
             .withProducerName(testStorage.getProducerName())
             .withConsumerName(testStorage.getConsumerName());
     }
+
+    public static KafkaClientsBuilder getInstantPlainClientBuilder(TestStorage testStorage) {
+        return instantClientBuilderBase(testStorage)
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(testStorage.getClusterName()));
+    }
+
+    public static KafkaClientsBuilder getInstantTlsClientBuilder(TestStorage testStorage) {
+        return instantClientBuilderBase(testStorage)
+            .withBootstrapAddress(KafkaResources.tlsBootstrapAddress(testStorage.getClusterName()));
+    }
```

Comment:
when you have plain and TLS, would it make sense to have SCRAM_SHA as well there?

Reply:
yes and no, I think that adding SCRAM_SHA makes sense from some viewpoint  but i found tls and plain as two most distinctive groups, i.e. you have either plain or tls data transmition.  As you know 
SCRAM_SHA can be applied on both of them so there would be really big overlap.